lib/reversetunnel/localsite.go

```diff
@@ -293,6 +297,63 @@ func (s *localSite) dialTunnel(dreq *sshutils.DialReq) (net.Conn, error) {
 	return conn, nil
 }
 
+// tryProxyPeering determines whether the node should try to be reached over
+// a peer proxy.
+func (s *localSite) tryProxyPeering(params DialParams) bool {
+	if s.peerClient == nil {
+		return false
+	}
+	if params.FromPeerProxy {
+		return false
+	}
+	if params.ConnType == "" || params.ConnType == types.ProxyTunnel {
+		return false
+	}
+
+	return true
+}
+
+// skipDirectDial determines if a direct dial attempt should be made.
+func (s *localSite) skipDirectDial(params DialParams) bool {
+	// Never direct dial when the client is already connecting from
+	// a peer proxy.
+	if params.FromPeerProxy {
+		return true
+	}
+
+	// Connections to application and database servers should never occur
+	// over a direct dial.
+	switch params.ConnType {
+	case types.AppTunnel, types.DatabaseTunnel:
+		return true
+	}
```

Comment:
IMO this switch statement should err on unexpected connection types, so we don't accidentally miss it if a new type is added that should be covered here.

Reply:
Doing another pass realized I never addressed this. Updated here https://github.com/gravitational/teleport/pull/10444/commits/065f59d1b26fcc92d7200a44f4c45c199a22b023